builder/rust/cargo/project_aspect.bzl

```diff
@@ -211,24 +211,47 @@ def _get_properties(target, ctx, source_files, crate_info):
         properties["edition"] = ctx.rule.attr.edition or _DEFAULT_RUST_EDITION
         entry_point_file = _entry_point_file(target, ctx, source_files)
         properties["entry.point.path"] = _src_relpath(target, ctx, entry_point_file)
-        properties["build.deps"] = ",".join(_crate_build_deps_info(crate_info))
+        if len(_crate_build_deps_info(crate_info)) > 0:
+            fail("Build deps support unimplemented")
+            #properties["build.deps"] = ",".join(_crate_build_deps_info(crate_info))
     for dep in _crate_deps_info(target, crate_info).items():
         properties["deps." + dep[0]] = dep[1]
     return properties
 
 def _crate_deps_info(target, crate_info):
     deps_info = {}
+
     for dependency in crate_info.deps:
         dependency_info = dependency[CrateInfo]
         if _is_universe_crate(dependency):
             location = "version={}".format(dependency_info.version)
-        else:
+        elif not _is_external_target(target) and _is_external_target(dependency):
             location = "path=../{}".format(dependency_info.crate_name)
+        else:
+            target_to_root = _package_relative_path_to_root(target.label)
```

Comment:
We relativise 'repository-local' crates using paths from the root instead of always pointing into the sandbox - this streamlines local development.